src/core_plugins/state_session_storage_redirect/public/index.js

```diff
@@ -0,0 +1,77 @@
+import 'ui/autoload/styles';
+import chrome from 'ui/chrome';
+import encodeUriQuery from 'encode-uri-query';
+import rison from 'rison-node';
+import { parse as parseUrl, format as formatUrl } from 'url';
+import { stringify as stringifyQuerystring } from 'querystring';
+
+const conservativeStringifyQuerystring = (query) => {
```

Comment:
I think this logic should be put in the `state_management` module next to src/ui/public/state_management/state_hashing/unhash_url.js, and it definitely needs tests.